modules/openapi-generator/src/main/java/org/openapitools/codegen/IJsonSchemaValidationProperties.java

```diff
@@ -213,4 +217,49 @@ default void setTypeProperties(Schema p) {
             setIsAnyType(true);
         }
     }
+
+    /**
+     * @return basic type - no generics supported.
+     */
+    default String getBaseType() {
+        return null;
+    };
+
+    /**
+     * @return complex type that can contain type parameters - like {@code List<Items>} for Java
+     */
+    default String getComplexType() {
+        return getBaseType();
+    };
+
+    /**
+     * Recursively collect all necessary imports to include so that the type may be resolved.
+     *
+     * @param includeContainerTypes whether or not to include the container types in the returned imports.
+     * @return all of the imports
+     */
+    default Set<String> getImports(boolean includeContainerTypes) {
+        Set<String> imports = new HashSet<>();
+        if (this.getComposedSchemas() != null) {
+            CodegenComposedSchemas composed = (CodegenComposedSchemas) this.getComposedSchemas();
+            List<CodegenProperty> allOfs =  composed.getAllOf() == null ? Collections.emptyList() : composed.getAllOf();
+            List<CodegenProperty> oneOfs =  composed.getOneOf() == null ? Collections.emptyList() : composed.getOneOf();
+            Stream<CodegenProperty> innerTypes = Stream.concat(allOfs.stream(), oneOfs.stream());
+            innerTypes.flatMap(cp -> cp.getImports(includeContainerTypes).stream()).forEach(s -> imports.add(s));
+        } else if (includeContainerTypes || !(this.getIsArray() || this.getIsMap())) {
+            // Composed types shouldn't be added directly because types like `Cat | Dog` won't have resolvable imports.
+            imports.add(this.getComplexType());
+            imports.add(this.getBaseType());
```

Comment:
This code is for our base case, can you move it above our first composedschema if?
```
if (this.getComplexType() != null) {
  // Base case: referenced components need to be imported
  imports.add(this.getComplexType());
}
```

Reply:
Conversely, if I don't add baseType here, I get 130 changed sample files where key imports are removed. Some languages apparently need to import string - like in cpp they import `QString` and `QDateTime`. I wasn't sure why these aren't included, but my goal was to be as non-destructive as possible while adding what I needed. So I left the base type import in. You can see that this was brought in directly in the old code here on line 5270 in `DefaultCodegen.java`